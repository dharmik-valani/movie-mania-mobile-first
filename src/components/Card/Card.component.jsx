import React from 'react'
import Playicon from '../../assets/Card/playicon.svg'
import Imagecomponent from '../Image/Image.component'

const Card = (props) => {
  return (
    <div className='bg-[#FAFAFA] p-5' data-testid="Play">
        <div className='flex gap-2 items-center'>
            <Imagecomponent src={Playicon} />
            <div>Play trailer {props?.index}</div>
        </div>
    </div>
  )
}

export default Card