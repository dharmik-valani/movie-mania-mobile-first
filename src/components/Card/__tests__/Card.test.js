import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "../Card.component.jsx";

describe("Card", () => {
  let originalFetch;

  beforeEach(() => {
    originalFetch = global.fetch;
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve([
            {
              index: 1,
              id: 1,
              title: "Play",
              completed: false,
            },
          ]),
      })
    );
  });

  afterEach(() => {
    global.fetch = originalFetch;
  });

  it("Renders Card component", async () => {
    let data = {
      index: 1,
      name: "hello"
    };
    render(<Card index={data.index} />);
    const linkElement = await screen.getByTestId('Play');
    expect(linkElement).toBeInTheDocument();
  });

  //  SNAP SHOT
  test("matches card snapshot", async () => {
    let data = {
      index: 1,
      name: "hello"
    };
    const tree = await renderer.create(<Card index={data.index} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
