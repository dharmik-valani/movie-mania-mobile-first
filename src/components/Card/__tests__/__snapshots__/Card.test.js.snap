// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches card snapshot 1`] = `
<div
  className="bg-[#FAFAFA] p-5"
  data-testid="Play"
>
  <div
    className="flex gap-2 items-center"
  >
    <img />
    <div>
      Play trailer 
      1
    </div>
  </div>
</div>
`;
